### DISCUSSION TITLE
Lesson 1: MNIST_Syft_Data_Owner notebook yields error for get_root_data_path()

### DISCUSSION THREAD

**Discussion by cbs-rschreijen** [OpenMined/courses#294] [Course: Foundations of Private Computation] [answered]
Running the first cell in the /duet_mnist/MNIST_Syft_Data_Owner.ipynb yields the following error:

---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-4-965c4bd46a83> in <module>
      6 urllib.request.install_opener(opener)
      7 import torchvision
----> 8 torchvision.datasets.MNIST(get_root_data_path(), train=True, download=True)
      9 torchvision.datasets.MNIST(get_root_data_path(), train=False, download=True)

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/util.py in get_root_data_path()
    595 
    596     data_dir = here / "data"
--> 597     os.makedirs(data_dir, exist_ok=True)
    598     return data_dir

~/opt/anaconda3/envs/conda-env/lib/python3.8/os.py in makedirs(name, mode, exist_ok)
    221             return
    222     try:
--> 223         mkdir(name, mode)
    224     except OSError:
    225         # Cannot rely on checking for EEXIST, since the operating system

OSError: [Errno 30] Read-only file system: '/data'

**Comment by znreza** [comment, marked as answer]
Hi @cbs-rschreijen this issue was reported by some others as well. Have you installed syft from https://github.com/OpenMined/PySyft@dev#egg=syft link? If yes then uninstall and re-install syft. It should work. Here is the PR for the change https://github.com/OpenMined/PySyft/pull/5312.

If you are in OpenMined slack, you can find how others solved it [here](https://openmined.slack.com/archives/C6EEFN3A8/p1616016761029000).

**Reply by cbs-rschreijen** [reply]
Thank you Zareen, uninstalling and reinstalling PySyft (effectively upgrading version again) fixes the probleem as you mentioned.

Unfortunately the http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz download is unavailable right now, but this is not fixable, just have to wait until it becomes available again.

How can I join OpenMined Slack? Is it by invitation only?

**Reply by znreza** [reply]
Hi @cbs-rschreijen please join at [openmined.slack.com](openmined.slack.com). It is open for all!

**Reply by cbs-rschreijen** [reply]
It turns out you need to use this URL https://openmined.slack.com/join/shared_invite/zt-nvhqkubp-haDCpXSasR3RbvJySIza8Q#/  to join openmined Slack. The link you provided does not allow sign-up, only sign-in.

Thanks again.
